### PR TITLE
feat(madoca): real-time MADOCA-PPP PPP-AR/IONO — wire L6D iono into mrtk run (SBF)

### LINKED ISSUE #223: feat(madoca): real-time PPP-AR/IONO — wire MADOCA-PPP L6D ionospheric augmentation into mrtk run
## Summary

Bring **PPP-AR/IONO** (MADOCA-PPP ionosphere-augmented integer PPP-AR) to the real-time path (`mrtk run`). Today this mode is **post-processing only**: the MADOCA-PPP L6E SSR (orbit/clock/code-bias/phase-bias) is decoded in real time, but the **L6D wide-area ionospheric augmentation** is not wired into the RT server, so `ionosphere = est-stec` + `iono_correction = on` cannot constrain the ionosphere states in `mrtk run`.

### Current RT capability (for reference)

| Mode | RT | Notes |
|------|:--:|------|
| PPP Float | ✅ | L6E SSR decoded in RT |
| PPP-AR | ✅ | uncombined (`est-stec`) + AR, no iono augmentation |
| **PPP-AR/IONO** | ❌ | needs L6D iono augmentation — **this issue** |

Supported systems are unchanged by this work: Float = GPS/GLONASS/Galileo/BeiDou/QZSS; AR = GPS/Galileo/QZSS/BeiDou (GLONASS float-only).

## Root cause — correction-path asymmetry

| Correction | Signal / PRN | Decoder | PP | RT |
|------------|--------------|---------|:--:|:--:|
| MADOCA SSR (orbit/clock/bias) | L6E | `input_qzssl6e` → `decode_qzss_l6emsg` | ✅ | ✅ `src/stream/mrtk_rtksvr.c:389`; SBF 4069 → `src/data/rcv/mrtk_rcv_septentrio.c:1211` |
| MADOCA iono augmentation | **L6D / QZS PRN 200,201** | `input_qzssl6d` → `decode_qzss_l6dmsg` | ✅ `src/pos/mrtk_postpos.c:719` | ❌ **not wired** |

- The RT server has **no L6D iono decode path** and never allocates `svr->nav.pppiono`. `ppp_iono.c` early-returns on `!nav->pppiono` (`src/pos/mrtk_ppp_iono.c:179`), so an `est-stec` RT run simply estimates the ionosphere freely (= plain PPP-AR), without the L6D constraint.
- Post-processing reads pre-split per-PRN `.200.l6` / `.201.l6` files (`src/pos/mrtk_postpos.c:1120`) and allocates `pppiono` when found (`:1150`). That file-based discrimination does not transfer to a live stream.

## Proposed approach

Mirror the existing real-time SSR-context pattern already used for Galileo HAS (`svr->has`) and BeiDou PPP-B2b (`svr->b2b`) — see `src/stream/mrtk_rtksvr.c:585` (decode + copy into `svr->nav.ssr_ch`) and `:1264` (allocate by `prcopt->correction`).

1. Add `mdcl6d_t *mdciono` to `rtksvr_t` (per-PRN array, `MIONO_MAX_PRN = 3`) and allocate `svr->nav.pppiono` in `rtksvrstart()`, gated on `correction == qzs-madoca && ionosphere == est-stec && iono_correction == on`. Free both in `rtksvrfree()`.
2. In `decoderaw()`, feed L6D iono bytes to `input_qzssl6d` / `decode_qzss_l6dmsg`; on `ret == 10`, copy `mdcl6d->re` into `svr->nav.pppiono->re[rid]` (the exact copy done by `update_qzssl6d()` in postpos).

Input has two sub-cases:

- **SBF single stream (mosaic; preferred):** L6E (block 4069) and L6D (block 4270) arrive in the same SBF correction stream. Today block 4270 is routed **straight to the CLAS decoder** (`src/data/rcv/mrtk_rcv_septentrio.c:1220`). MADOCA iono L6D must be discriminated from CLAS L6D by **`vid == 2` and `prn ∈ {200, 201}`** and branched to the iono decoder. No extra stream slot needed.
- **Raw L6D stream:** define a new `STRFMT_L6D` and route it to `input_qzssl6d`. RT has only 3 input slots and L6E SSR occupies `corr`; the L6D iono stream would take the `base` slot (same constraint already accepted for 2-channel CLAS).

## Implementation tasks

- [ ] `rtksvr_t`: add `mdcl6d_t *mdciono` (heap, `MIONO_MAX_PRN`); allocate/free with `pppiono`
- [ ] `rtksvrstart()`: allocate `pppiono` + `mdciono` under the gate above
- [ ] `decoderaw()`: L6D iono decode + region copy into `nav.pppiono->re[rid]`, with PRN→region demux
- [ ] Septentrio SBF: discriminate MADOCA iono L6D (`vid==2`, PRN 200/201) from CLAS L6D without regressing CLAS
- [ ] (raw input) define `STRFMT_L6D`, parser dispatch, TOML format mapping
- [ ] Config: consult `iono_correction` in the RT path; expose L6D iono stream in `conf/malib/rtkrcv*.toml`; update `resolve_correction()` validity matrix if needed
- [ ] Age / IOD synchronization between L6E SSR and L6D iono (analogous to the CLAS trop-age gate)
- [ ] Handover / `alert` handling across PRN 200/201
- [ ] Docs + capability matrix update (RT PPP-AR/IONO now supported)

## Challenges / risks

1. **CLAS L6D coupling (primary risk).** SBF block 4270 is hard-wired to CLAS — a path repeatedly implicated in handover-freeze regressions (#197, #205, #218). The new discriminator must not perturb CLAS behavior.
2. **Stream-slot scarcity.** Raw L6D input consumes the `base` slot, mutually exclusive with base-station RTK. SBF single-stream is the realistic path.
3. **No RT regression harness.** `rtkrcv_rt` is environment-broken on the maintainer machine; automated RT regression is hard. Validation likely needs a scripted replay compared against the PP reference.
4. **Test data.** Need an SBF capture (or two raw streams) containing **both** L6E SSR and L6D iono (PRN 200/201). Unverified whether the bundled `MALIB_OSS_data_l6e_*.sbf` includes block 4270 / PRN 200,201; if not, a fresh hardware capture is required (adds lead time).
5. **Asynchronous arrival.** PP reads SSR and iono together; RT receives them on independent cadences and must age-gate to keep them time/IOD-consistent.

## Cost estimate

**~1 week (5–7 dev-days), medium risk.** The numerical core is fully reused from the post-processing decoder (low algorithm risk); the cost and risk concentrate in RT plumbing that touches the sensitive CLAS L6D path and the scarce stream-slot architecture.

| Work | Est. |
|------|:----:|
| RT decode wiring (mdciono/pppiono alloc/free, region copy) | 1–2 d |
| SBF L6D iono-vs-CLAS discrimination | 1 d |
| Config plumbing (`ionocorr` in RT, TOML, `resolve_correction`) | 0.5 d |
| Test data prep + numerical match vs PP reference | 1–2 d |
| Edge cases (age gate, handover, alert) | 1 d |

## Acceptance criteria

- [x] `mrtk run` with MADOCA-PPP `correction = qzs-madoca`, `ionosphere = est-stec`, `iono_correction = on` populates `nav.pppiono` and applies L6D iono constraints in real time
- [x] RT solution matches the post-processing PPP-AR/IONO reference for the same epochs within tolerance (cf. `madocalib_pppar_ion` PP reference)
- [ ] No regression in real-time CLAS (handover behavior unchanged) or in MADOCA Float / PPP-AR RT
- [ ] Docs / capability matrix updated

## References

- Post-processing iono path: `src/pos/mrtk_postpos.c:447-514`, `:719`, `:1120-1151`
- L6D decoder: `src/madoca/mrtk_madoca_iono.c` (`input_qzssl6d`, `decode_qzss_l6dmsg`; PRN 200/201/197 at `:356`)
- RT SSR-context analogs: `src/stream/mrtk_rtksvr.c:565` (HAS), `:585` (B2b), `:1255-1266` (allocation)
- Iono application: `src/pos/mrtk_ppp_iono.c:168-210`
- Related: #86 (UBX input for MADOCA-PPP RT)

### PR DESCRIPTION
**Pull request by h-shiono**
## Summary

Brings **PPP-AR/IONO** (MADOCA-PPP L6D wide-area ionospheric augmentation) to the real-time path (`mrtk run`). Previously this mode was **post-processing only**: L6E SSR was decoded in real time, but the **L6D wide-area iono augmentation (QZS PRN 200/201)** was never wired into the RT server, so `ionosphere = est-stec` + `iono_correction = on` ran as plain PPP-AR without the iono constraint.

**SBF-first** (mosaic-G5): a single SBF stream carries observations, L6E SSR (QZSRawL6E 4271) and L6D iono (QZSRawL6D 4270, PRN 200/201) in one slot — no extra stream slot needed.

Closes #223.

## What changed

- **`mrtk_rcv_septentrio.c`** — in the `Source=1` (L6D) branch, return a dedicated code **16** for PRN 200/201/197 (MADOCA iono) instead of 10 (CLAS redirect). The discriminator is the **PRN**: 200/201 carry the iono "Technology Demonstration" and are disjoint from CLAS (193–199) per IS-QZSS-MDC-004 Table 3-1.
- **`rtksvr_t`** — add per-PRN `mdciono` contexts (`MIONO_MAX_PRN`) and allocate `nav.pppiono`, gated on `correction = qzs-madoca` + `iono_correction` + `ionosphere = est-stec`; free both in `rtksvrfree` (pppiono was never freed in the RT path before).
- **`decoderaw`** — on `ret==16`, feed the L6D frame to `input_qzssl6d` and copy the decoded region into `nav.pppiono->re[rid]`, mirroring `update_qzssl6d()` in `mrtk_postpos.c`. The GPST week is anchored lazily from the first stream time (the CLAS `week_ref` pattern) so recorded-stream replay decodes in the correct week. The `update_svr` skip guard is extended to `ret==16`.
- **`conf/madocalib/rtkrcv_madoca_pppar_iono.toml`** — real-time single-SBF sample.
- **docs** — real-time PPP-AR/IONO subsection in the MADOCA-PPP quickstart.

`resolve_correction()` needs no change — it already permits `qzs-madoca` + `est-stec` + `iono_correction`.

## Why no CLAS regression

In MADOCA-PPP mode `svr->clas == NULL`, so the new path never touches CLAS. In CLAS mode `svr->mdciono == NULL`, and PRN 200/201 were already rejected by the CLAS decoder (they are not CLAS PRNs), so routing them to code 16 instead of 10 changes nothing functionally on the sensitive L6D path (#197/#205/#218).

## Validation

- **Offline decode** (`G5P3173a.sbf`, mosaic-G5): 3660 L6D iono frames → 366 complete decodes → `nav.pppiono` regions {1,3,4} populated (100% valid), alongside 89 L6E SSR satellites from the same stream.
- **Regression**: `ctest` 97/98. The single failure is the pre-existing LAPACK-tolerance `madocalib_pppar_ion_check` (3D RMS 1.62 cm vs 0.5 cm tol, fix-rate delta +0.00%); the post-processing iono path is unchanged. The `rtkrcv_rt` perennial passed this run.
- **Live real-device** (mosaic-G5 over TCP, Tokyo): the new path fires in real time (trace `L6D iono: prn=200 rid=1/3/4`), L6E SSR age ≈ 1 s, and **PPP-AR/IONO reached a fixed solution (Q=1) ~57 s after start**, holding continuous Q=1 at **sdn ≈ 1.2 cm / sde ≈ 1.2 cm / sdu ≈ 3.4 cm** (ns=30). The fast fix is the L6D iono augmentation working as intended.
- Build clean; clang-format 21.1.6 / taplo 0.10.0 clean.

## Follow-up (separate PR)

- **u-blox D9C path** — apply the same `return 16` in `decode_rxmqzssl6()` (`mrtk_rcv_ublox.c`, `msg==0` / PRN 200/201) and reuse the rtksvr block. Planned **together with #86** (UBX MADOCA-PPP L6E correction input in `mrtk run`), since both complete the D9C real-time MADOCA path.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/stream/mrtk_rtksvr.c [hunk 1]
```diff
@@ -565,6 +567,51 @@ static int decoderaw(rtksvr_t* svr, int index) {
...
+                    gtime_t t = svr->raw[index].time.time ? svr->raw[index].time : svr->raw[0].time;
+                    if (t.time) {
+                        int m;
+                        init_miono(t);
+                        for (m = 0; m < MIONO_MAX_PRN; m++) {
+                            svr->mdciono[m].time = t;
+                            svr->mdciono[m].nbyte = 0;
+                            svr->mdciono[m].re.rvalid = 0;
+                            strncpy(svr->mdciono[m].opt, svr->rtk.opt.pppopt, sizeof(svr->mdciono[m].opt) - 1);
+                        }
```
[the remark this answers is not in the document]
Comment (h-shiono) [reply src/stream/mrtk_rtksvr.c:594]: Good catch — fixed in 312cac6 by explicitly terminating `opt[sizeof(opt)-1] = '\0'` after the `strncpy`. (In practice the buffer is `calloc`'d so the last byte starts at 0, but the explicit terminator makes the L6D decoder's `strstr()`/`atoi()` over `mdcl6d->opt` safe regardless of `pppopt` length.)
